db/db_impl_open.cc

```diff
@@ -1169,8 +1170,13 @@ Status DBImpl::Open(const DBOptions& db_options, const std::string& dbname,
       if (!s.ok()) {
         break;
       }
+      int num_levels = cfd->current()->storage_info()->num_levels();
+      max_num_levels = num_levels > max_num_levels ? num_levels : max_num_levels;
     }
   }
+  // TODO(Zhongyi): need to initialize this in all threads that might need
+  // access to PerfContextByLevel
+  perf_context.EnablePerLevelPerfContext(max_num_levels);
```

Comment:
This is wrong place to do the thing. Open() may not be done in the same thread as query.

And we cannot initialize for all threads. It's bad. People need to enable it their for their thread.

Also be aware that perf_context is not per DB or per CF. It's shared across all DBs and all CFs. Getting max level for one CF and allocate based on it is not a good idea. Maybe we should consider to allocate a fixed number, say 16. All levels 15 or up fall into level 15. Or maybe we can even think of making it std::map from level to struct.